### PR TITLE
MRPA Tests

### LINKED ISSUE #35: Tests: Test MRPA algorithm

### PR DESCRIPTION
**Pull request by KarmaKamikaze**
**Summary of added content:**

- Tweaks to the MRPA algorithm
  - Remove duplicates that may occur when the resolution scale is very small
- Tests
  - Tests for MRPA algorithm
  - Tests for Node template
  - Tests for Trajectory model

**Relevant files for review:**

Familiarize yourselves with the CMakeLists files
- `src/simp-algorithms/MRPA.hpp`
- `src/simp-algorithms/MRPA.cpp`
- `tests/mrpa_test.cpp`
- `tests/node_test.cpp`
- `tests/trajectory_test.cpp`

This pull request closes #35

### CODE REVIEW

File: tests/mrpa_test.cpp [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+        return loc1.timestamp == loc2.timestamp
+               && loc1.longitude == loc2.longitude
+               && loc1.latitude == loc2.latitude;
+    };
+
+    data_structures::Trajectory small{};
+    data_structures::Trajectory medium{};
+    data_structures::Trajectory large{};
+
+    test_trajectories() {
```
Comment (Kristoffermg) [inline tests/mrpa_test.cpp:17]: Man kunne eventuelt lave 3 for loops til det her som sætter tilfældige værdier ind

File: tests/mrpa_test.cpp [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+        for (const auto& r : res) {
+            CHECK(tt.compare_locations(r.locations.front(), first));
+            CHECK(tt.compare_locations(r.locations.back(), last));
+        }
+    }
+}
+
+TEST_CASE("MRPA - Simplification of a straight line yields only the first and last locations") {
+    auto straight_boi = data_structures::Trajectory{};
```
Comment (Kristoffermg) [inline tests/mrpa_test.cpp]: Straight boi? 🤔

**Review by Kristoffermg — CHANGES_REQUESTED** [review]

**Review by Flaiizor — APPROVED** [review]
Det er tests 😐🤷‍♀️🤷🏿‍♀️

File: tests/mrpa_test.cpp (hunk 2 above)
Comment (KarmaKamikaze) [reply tests/mrpa_test.cpp]: Han er død :skull:

**Review by Kristoffermg — APPROVED** [review]
Massere gode tests

**Pull request merged**